src/app/admin-page/student-form/student-form.component.ts

```diff
@@ -0,0 +1,152 @@
+import {Component, Input, OnInit} from '@angular/core';
+import {Observable} from 'rxjs';
+import {CustomError} from '../../shared/customError';
+import {FormBuilder, FormControl, FormGroup, Validators} from '@angular/forms';
+import {Store} from '@ngrx/store';
+import {
+  getHasStudentAddFailed,
+  getHasStudentEditFailed,
+  getIsStudentsLoaded,
+  getIsStudentsLoading,
+  getStudentById,
+  getStudentEditId,
+  getStudentsError
+} from '../../store/students/students.selectors';
+import {addStudent, editStudent} from '../../store/students/students.actions';
+import {switchMap} from 'rxjs/operators';
+import {Student} from '../../shared/student';
+import {StudentFormType} from './StudentFormType';
+import {ProfilePictureService} from '../../service/profile-picture.service';
+
+const namePattern = '^[a-zA-ZĄąČčĘęĖėĮįŠšŲŲūŪŽž]*$';
+const occupationPattern = '^[a-zA-ZĄąČčĘęĖėĮįŠšŲųūŪŽž_-\\s]*$';
+const noMultipleSpacesPattern = '(?:(?![ ]{2}).)+';
+
+@Component({
+  selector: 'app-student-form',
+  templateUrl: './student-form.component.html',
+  styleUrls: ['./student-form.component.scss']
+})
+export class StudentFormComponent implements OnInit {
+  @Input() type!: StudentFormType;
```

Comment:
Since this component can only have two scenarios, a boolean would be a more simple solution :)
```suggestion
  @Input() isEdItView: boolean;
```
```if (isEditView) {} else {}``` ;) same for template ```*ngIf="isEditView; then xxx; else xxx"```
While you are assigning the same values to different variables and creating unnecessary boilerplate code :)